pkg/runtime/wfengine/workflow.go

```diff
@@ -98,13 +112,26 @@ func (wf *workflowActor) InvokeReminder(ctx context.Context, actorID string, rem
 	wfLogger.Debugf("invoking reminder '%s' on workflow actor '%s'", reminderName, actorID)
 
 	// Workflow executions should never take longer than a few seconds at the most
-	timeoutCtx, cancelTimeout := context.WithTimeout(ctx, 30*time.Second) // TODO: Configurable
+	timeoutCtx, cancelTimeout := context.WithTimeout(ctx, wf.defaultTimeout)
 	defer cancelTimeout()
 	if err := wf.runWorkflow(timeoutCtx, actorID, reminderName, data); err != nil {
-		return err
+		if err == timeoutCtx.Err() {
```

Comment:
Here too note the error should be checked against `errors.Is(err, context.Canceled)` and `errors.Is(err, context.DeadlineExceeded)`

Reply:
Updated to use `errors.Is(err, context.DeadlineExceeded)` but not `errors.Is(err, context.Canceled)` since we never explicitly cancel.